app/src/main/java/net/gsantner/markor/util/DocumentIO.java

```diff
@@ -77,7 +88,26 @@ public static synchronized Document loadDocument(Context context, Bundle argumen
         } else if (filePath.isFile() && filePath.canRead()) {
             // Extract content and title
             document.setTitle(filePath.getName());
-            document.setContent(FileUtils.readTextFileFast(filePath));
+            String content;
```

Comment:
I will do later a checkup of this whole logic. It looks to me like it could cause lots of issues, especially  
"my whole text got deleted. help how can I get it back?"

Reply:
The critical part is the load function. If the text can't be decrypted, I return an empty string. If you then store this, the text is gone. But I have no idea how to prevent this.

Reply:
Ok then it should not be a problem, as Markor is built to NOT save empty string to file. Still, could you please try some bad strings and see what happens for real?

Reply:
Test an improved version with wrong password. No content will become lost.